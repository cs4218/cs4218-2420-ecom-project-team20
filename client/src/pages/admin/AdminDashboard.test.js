import { render } from "@testing-library/react";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { useAuth } from "../../context/auth";
import React from "react";
import AdminDashboard from "./AdminDashboard";
import "@testing-library/jest-dom";

// Mock Layout component
jest.mock("../../components/Layout", () =>
  jest.fn(({ children }) => <div>{children}</div>)
);

// Mock AdminMenu component
jest.mock("../../components/AdminMenu", () =>
  jest.fn(() => <div>Admin Menu</div>)
);

// Mock useAuth component
jest.mock("../../context/auth", () => ({
  useAuth: jest.fn(),
}));

describe("AdminDashboard Component", () => {
  const mockAuth = {
    user: {
      name: "John Doe",
      email: "admin@example.com",
      phone: "123-456-7890",
    },
  };

  beforeEach(() => {
    useAuth.mockReturnValue([mockAuth]);
  });

  it("renders Layout component", () => {
    const { container } = render(
      <MemoryRouter>
        <AdminDashboard />
      </MemoryRouter>
    );

    // Check if Layout's class is in the document
    expect(container.querySelector(".container-fluid")).toBeInTheDocument();
  });

  it("renders AdminDashboard component correctly", () => {
    useAuth.mockReturnValue([
      {
        user: {
          name: "John Doe",
          email: "admin@example.com",
          phone: "123-456-7890",
        },
      },
    ]);

    const { container } = render(
      <MemoryRouter initialEntries={["/dashboard/admin"]}>
        <Routes>
          <Route path="/dashboard/admin" element={<AdminDashboard />} />
        </Routes>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot(); // âœ… Snapshot test
  });

  it("renders AdminMenu inside AdminDashboard", () => {
    // Mock the auth state
    useAuth.mockReturnValue([
      {
        user: {},
      },
    ]);
    const { getByText } = render(
      <MemoryRouter initialEntries={["/dashboard/admin"]}>
        <Routes>
          <Route path="/dashboard/admin" element={<AdminDashboard />} />
        </Routes>
      </MemoryRouter>
    );
    expect(getByText("Admin Menu")).toBeInTheDocument();
  });

  it("displays admin user details correctly", () => {
    useAuth.mockReturnValue([
      {
        user: {
          name: "John Doe",
          email: "admin@example.com",
          phone: "123-456-7890",
        },
      },
    ]);

    const { getByText } = render(
      <MemoryRouter initialEntries={["/dashboard/admin"]}>
        <Routes>
          <Route path="/dashboard/admin" element={<AdminDashboard />} />
        </Routes>
      </MemoryRouter>
    );

    expect(getByText("Admin Name : John Doe")).toBeInTheDocument();
    expect(getByText("Admin Email : admin@example.com")).toBeInTheDocument();
    expect(getByText("Admin Contact : 123-456-7890")).toBeInTheDocument();
  });

  it("handles missing user data gracefully", () => {
    useAuth.mockReturnValue([{}]); // No user data

    const { getByText } = render(
      <MemoryRouter initialEntries={["/dashboard/admin"]}>
        <Routes>
          <Route path="/dashboard/admin" element={<AdminDashboard />} />
        </Routes>
      </MemoryRouter>
    );

    expect(getByText("Admin Name :")).toBeInTheDocument();
    expect(getByText("Admin Email :")).toBeInTheDocument();
    expect(getByText("Admin Contact :")).toBeInTheDocument();
  });

  it("handles null auth data gracefully", () => {
    useAuth.mockReturnValue(null); // Simulate missing auth context

    const { getByText } = render(
      <MemoryRouter initialEntries={["/dashboard/admin"]}>
        <Routes>
          <Route path="/dashboard/admin" element={<AdminDashboard />} />
        </Routes>
      </MemoryRouter>
    );

    expect(getByText("Admin Name :")).toBeInTheDocument();
    expect(getByText("Admin Email :")).toBeInTheDocument();
    expect(getByText("Admin Contact :")).toBeInTheDocument();
  });
});
