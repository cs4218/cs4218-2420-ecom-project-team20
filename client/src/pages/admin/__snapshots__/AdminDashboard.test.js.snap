// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard Component renders AdminDashboard component correctly 1`] = `
<div>
  <div>
    <div
      class="container-fluid m-3 p-3"
    >
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div>
            Admin Menu
          </div>
        </div>
        <div
          class="col-md-9"
        >
          <div
            class="card w-75 p-3"
          >
            <h3>
               Admin Name : 
              John Doe
            </h3>
            <h3>
               Admin Email : 
              admin@example.com
            </h3>
            <h3>
               Admin Contact : 
              123-456-7890
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
